app/src/main/java/de/danoeh/antennapod/fragment/DiscoveryFragment.java

```diff
@@ -221,4 +181,81 @@ private void loadToplist(String country) {
                     });
         }
     }
+
+    @Override
+    public boolean onMenuItemClick(MenuItem item) {
+        if (super.onOptionsItemSelected(item)) {
+            return true;
+        }
+        final int itemId = item.getItemId();
+        if (itemId == R.id.discover_hide_item) {
+            item.setChecked(!item.isChecked());
+            if (item.isChecked()) {
+                previousCountryCode = countryCode;
+                countryCode = ItunesTopListLoader.DISCOVER_HIDE_FAKE_COUNTRY_CODE;
+            } else {
+                countryCode = previousCountryCode;
+            }
+            prefs.edit().putString(ItunesTopListLoader.PREF_KEY_COUNTRY_CODE, countryCode).apply();
+
+            EventBus.getDefault().post(new DiscoveryDefaultUpdateEvent());
+            loadToplist(countryCode);
+            return true;
+        } else if (itemId == R.id.discover_countries_item) {
+
+            LayoutInflater inflater = getLayoutInflater();
+            View selectCountryDialogView = inflater.inflate(R.layout.select_country_dialog, null);
+            AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+            builder.setView(selectCountryDialogView);
+
+            List<String> countryCodeArray = new ArrayList<>(Arrays.asList(Locale.getISOCountries()));
+            Map<String, String> countryCodeNames = new HashMap<>();
+            Map<String, String> countryNameCodes = new HashMap<>();
+            for (String code : countryCodeArray) {
+                Locale locale = new Locale("", code);
+                String countryName = locale.getDisplayCountry();
+                countryCodeNames.put(code, countryName);
+                countryNameCodes.put(countryName, code);
+            }
+
+            List<String> countryNamesSort = new ArrayList<>(countryCodeNames.values());
+            Collections.sort(countryNamesSort);
+
+            ArrayAdapter<String> dataAdapter =
+                    new ArrayAdapter<>(this.getContext(), R.layout.list_item, countryNamesSort);
+            TextInputLayout textInput = selectCountryDialogView.findViewById(R.id.country_text_input);
+            MaterialAutoCompleteTextView editText = (MaterialAutoCompleteTextView) textInput.getEditText();
+            editText.setAdapter(dataAdapter);
+            if (countryCode.equals(ItunesTopListLoader.DISCOVER_HIDE_FAKE_COUNTRY_CODE)) {
+                editText.setText(countryCodeNames.get(previousCountryCode));
```

Comment:
Why do you store the previous code? It's only in a variable anyway, so it doesn't really help?

Reply:
I use this variable in order to restore country user previously selected when they uncheck "Hide" checkbox, and to show previously selected country in drop-down list instead of empty string

Reply:
Also made this variable stored in prefs

Reply:
Instead of storing (current, previous) country codes where the current one can be `DISCOVER_HIDE_FAKE_COUNTRY_CODE`, I think it would be a lot more clear to store (country, hidden). Then we need to store one string and one boolean and don't need to move strings back and forth between the two variables.

Reply:
fixed